File: lab_4/lab_4.py
def main():
    try:
        a, b, n = float(input("a = ")), float(input("b = ")), int(input("n = "))
    except:
        print("Oh-oh, it seems the data you entered is invalid ^_^")
        return

    if n == 0:
        print("Sorry, but n cannot equal 0")
        return
    if a == b:
        print("Sorry, but a cannot equal b")
        return

    h = (b - a)/n

    for i in range(1, n + 1):
        result = a + i*h
        print(f"r{i} — {round(result, 3)}")

if __name__ == "__main__":
    main()

File: Lab4/Lab4.cpp
#include <iostream>

using namespace std;

int main()
{
	double a, b, h;
	int n;

	// data input
	cout << "a = "; cin >> a;
	cout << "b = "; cin >> b;
	cout << "n = "; cin >> n;
	
	if (n == 0) {
		cout << "Sorry, but n cannot equal 0";
		return 0;
	}
	if (a == b) {
		cout << "Sorry, but a cannot equal b\n";
		return 0;
	}

	h = (b - a) / n;

	for (int i = 1; i <= n; i++) {
		printf("r%i - %.3f\n", i, a + i * h);
	}

	return 0;
}